gcloud-java-bigquery/src/main/java/com/google/gcloud/spi/DefaultBigQueryRpc.java

```diff
@@ -0,0 +1,448 @@
+import com.google.api.services.bigquery.model.QueryResponse;
+import com.google.api.services.bigquery.model.Table;
+import com.google.api.services.bigquery.model.TableDataInsertAllRequest;
+import com.google.api.services.bigquery.model.TableDataInsertAllResponse;
+import com.google.api.services.bigquery.model.TableDataList;
+import com.google.api.services.bigquery.model.TableList;
+import com.google.api.services.bigquery.model.TableReference;
+import com.google.api.services.bigquery.model.TableRow;
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Iterables;
+
+import static com.google.gcloud.spi.BigQueryRpc.Option.MAX_RESULTS;
+import static com.google.gcloud.spi.BigQueryRpc.Option.PAGE_TOKEN;
+
+import com.google.gcloud.bigquery.BigQueryException;
+import com.google.gcloud.bigquery.BigQueryOptions;
+
+import java.io.IOException;
+import java.math.BigInteger;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class DefaultBigQueryRpc implements BigQueryRpc {
+
+  public static final String DEFAULT_PROJECTION = "full";
+  // see: https://cloud.google.com/bigquery/troubleshooting-errors
+  private static final Set<Integer> RETRYABLE_CODES = ImmutableSet.of(500, 502, 503, 504);
+  private final BigQueryOptions options;
+  private final Bigquery bigquery;
+
+  public DefaultBigQueryRpc(BigQueryOptions options) {
+    HttpTransport transport = options.httpTransportFactory().create();
+    HttpRequestInitializer initializer = options.httpRequestInitializer();
+    this.options = options;
+    bigquery = new Bigquery.Builder(transport, new JacksonFactory(), initializer)
+        .setRootUrl(options.host())
+        .setApplicationName(options.applicationName())
+        .build();
+  }
+
+  private static BigQueryException translate(IOException exception) {
+    BigQueryException translated;
+    if (exception instanceof GoogleJsonResponseException
+        && ((GoogleJsonResponseException) exception).getDetails() != null) {
+      translated = translate(((GoogleJsonResponseException) exception).getDetails());
+    } else {
+      translated =
+          new BigQueryException(BigQueryException.UNKNOWN_CODE, exception.getMessage(), false);
+    }
+    translated.initCause(exception);
+    return translated;
+  }
+
+  private static BigQueryException translate(GoogleJsonError exception) {
+    boolean retryable = RETRYABLE_CODES.contains(exception.getCode());
+    return new BigQueryException(exception.getCode(), exception.getMessage(), retryable);
+  }
+
+  @Override
+  public Dataset getDataset(String datasetId, Map<Option, ?> options) throws BigQueryException {
+    try {
+      return bigquery.datasets()
+          .get(this.options.projectId(), datasetId)
+          .setFields(FIELDS.getString(options))
+          .setQuotaUser(QUOTA_USER.getString(options))
```

Comment:
I added them as I was trying to have a comprehensive understanding of all possible options for each method. Since we ignored them in other services we can probably remove support for them (unless you want to add them to other modules - I don't mind doing it).